spec/controllers/items_controller_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'rails_helper'
+
+RSpec.describe ItemsController, type: :controller do
+  describe '#index' do
+    let!(:items) { create_list(:item, 7) }
+    context 'home' do
+      before { get :index }
+
+      it { expect(controller.items).to contain_exactly(*items[0..5]) }
+
+      it { expect(response).to render_template(:index) }
+    end
+
+    context 'search' do
+      let(:item) { create(:item) }
+      before { get :index, params: { q: { name_cont: item.name } } }
+
+      it { expect(controller.items).to contain_exactly(item) }
+    end
+  end
+
+  describe '#show' do
+    let(:item) { create(:item) }
+    let(:call_request) { get :show, params: { id: item.id } }
+
+    it_behaves_like 'an action rendering view'
+  end
+
+  describe '#destroy' do
+    let!(:item) { create(:item) }
+
+    context 'when logged in as admin' do
+      let!(:admin) { create(:admin) }
+      context 'when is no bidders' do
+        let(:call_request) { delete :destroy, params: { id: item.id } }
+        before { sign_in admin }
+
+        it_behaves_like 'an action destroying object'
+      end
+
+      context 'when is at least one bidder' do
+        let(:user) { create(:user) }
+        let!(:bid) { item.bids.create(user: user) }
+        before { sign_in admin }
+        let(:call_request) { delete :destroy, params: { id: item.id } }
```

Comment:
`call_request` can be move one level up and this will help to reduce code duplication (e.g.: to line 31)